src/components/Button/Button.js

```diff
@@ -190,19 +216,23 @@ Button.propTypes = {
    * Should the Button be disabled?
    */
   disabled: PropTypes.bool,
-  /**
-   * Type of button to render. For example, use 'primary' for confirmation button
-   * and 'secondary' for cancel button.
-   */
-  variant: PropTypes.oneOf(['primary', 'secondary']),
   /**
    * Button has a 'flat' variation, triggered with this prop.
    */
   flat: PropTypes.bool,
+  /**
+   * Renders a secondary button. Secondary buttons look the same for
+   * primary (default) and flat buttons.
+   */
+  secondary: PropTypes.bool,
```

Comment:
I like this :) An API like `<Button flat secondary/>` looks really clean and simple. I just wonder if we're gonna end up with funny combos, but yolo :D

Reply:
Hm yeah, there are funny combos. `<Button flat secondary>` will look just like `<Button secondary>` and I guess as we increase complexity of Button modifiers, there will be more. In Polaris you can make a [primary, destructive, outline button](https://polaris.shopify.com/components/actions/button#navigation). All you get is a destructive, outline button.